packages/api/src/main/java/com/incubator/edupayroll/controller/teacher/TeacherExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package com.incubator.edupayroll.controller.teacher;
+
+import com.incubator.edupayroll.service.teacher.TeacherNotFoundException;
+import com.incubator.edupayroll.util.response.Response;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice(assignableTypes = TeacherController.class)
+public class TeacherExceptionHandler {
+
+    @ExceptionHandler({TeacherNotFoundException.class})
+    public ResponseEntity<Response<?, TeacherErrorCode>> handleTeacherNotFoundException() {
+        return ResponseEntity
+                .status(HttpStatus.UNAUTHORIZED)
```

Comment:
bunu not found yapalım. kişinin erişimi olmadığı kaynakları engellemeyi access denied hatası ile globalde tutuyoruz zaten. access denied hatasının kullanımı service içinde göstereceğim. not found'u kişinin aradığı teacher gerçekten yoksa atmış olacağız.

ayrıca bu tarz controller exception handler'larında hatanın mesajları exception classından gelsin. ex.getMessage() gibi. o mesajda hatanın atıldığı yerde belirlesin. ama genellikle bunu exceptionun static generator metodunda yapıyoruz. `byUser` gibi